src/clients/SpokePoolClient.ts

```diff
@@ -688,8 +699,11 @@ export class SpokePoolClient extends BaseAbstractClient {
           realizedLpFeePct: dataForQuoteTime[index].realizedLpFeePct,
           destinationToken: this.getDestinationTokenForDeposit(rawDeposit),
           quoteBlockNumber: dataForQuoteTime[index].quoteBlock,
-          blockTimestamp: blocks[event.blockNumber]?.timestamp ?? (await event.getBlock()).timestamp,
         };
+        if (blocks !== undefined) {
+          (deposit as UBADepositWithBlock).blockTimestamp =
+            blocks[event.blockNumber]?.timestamp ?? (await event.getBlock()).timestamp;
```

Comment:
QQ: we've had issues in the past of `getBlock()` failing due to an undefined/null block. Should we be doing any form of type checking this far downstream or should we fully rely on the upstream error checks?

Reply:
I didn't want to change the logic at this point, what do you suggest?

Reply:
`null` blocks should now be prevented by this change at the provider layer: https://github.com/across-protocol/relayer-v2/commit/97fe05b883b062c3edd816f6b38b00ee06fe5ed2